hw30/index.js

```diff
@@ -0,0 +1,223 @@
+                selected.setSelected(true);
+            });
+
+        }
+
+        clearSelected() {
+            this.items.forEach((elem) => {
+                elem.setSelected(false);
+            });
+        }
+
+        getSelected() {
+            const selectedItems = this.items.filter((listItem) => listItem.isSelected);
+            return selectedItems.length ? selectedItems : null;
+        }
+
+        getListItem(target) {
+            return this.items.find((listItem) => listItem.element === target);
+        }
+
+        addListItem(text, position) {
+            const listItem = new ListItem(text);
+            const listItemElement = this.createListElement(listItem);
+
+            if (position === 'start') {
+                this.items.unshift(listItem);
+                this.element.insertAdjacentElement('afterbegin', listItemElement);
+            } else {
+                this.items.push(listItem);
+                this.element.insertAdjacentElement('beforeend', listItemElement);
+            }
+            return listItem;
+        }
+
+        removeListItem(listItem) {
+            listItem.remove();
+            this.items = this.items.filter((item) => item !== listItem);
+        }
+
+        createListElement(item) {
+            const listItemElement = document.createElement('li');
+            listItemElement.innerText = item.text;
+            item.setElement(listItemElement);
+            return listItemElement;
+        }
+
+        renderList(container) {
+            const list = this.element;
+            container.appendChild(list);
+        }
+
+        get listItems() {
+            return this.items;
+        }
+    }
+
+    class Menu {
+        constructor(elem, taskInput) {
+            this.input = taskInput;
+            this._elem = elem;
+            elem.onclick = this.onClick.bind(this);
+        }
+
+        validateTask() {
+            const value = this.input.value;
+            if (!value) alert('Please specify the task');
+            return value;
+        }
+
+
+        start() {
+            if (this.validateTask()) {
+                list.addListItem(this.input.value, 'start');
+                this.input.value = '';
+            }
+        }
+
+        end() {
+            if (this.validateTask()) {
+                list.addListItem(this.input.value);
+                this.input.value = '';
+            }
+        }
+
+        delete() {
+            const itemsToRemove = list.getSelected();
+            if (itemsToRemove) {
+                itemsToRemove.forEach((listItem) => list.removeListItem(listItem));
+            }
+        }
+
+        sort() {
+            const itemsToMove = list.getSelected();
+            if (itemsToMove) {
+                itemsToMove.forEach((listItem) => list.removeListItem(listItem));
+            }
+
+            itemsToMove.reverse().forEach((listItem) => list.addListItem(listItem.text, 'start').setSelected(true));
+        }
```

Comment:
Why do you have 2 loops here? And why second loop is out of the condition?  You will get an error if you call this method without the elements selected